data/trie/sync.go

```diff
@@ -266,34 +304,14 @@ func (ts *trieSyncer) requestNodes() uint32 {
 		}
 	}
 	ts.trieSyncStatistics.SetNumMissing(ts.rootHash, len(hashes))
-	ts.requestHandler.RequestTrieNodes(ts.shardId, hashes, ts.topic)
+	if len(hashes) > 0 {
+		ts.requestHandler.RequestTrieNodes(ts.shardId, hashes, ts.topic)
+	}
 	ts.mutOperation.RUnlock()
 
 	return numUnResolvedNodes
 }
 
-func (ts *trieSyncer) trieNodeIntercepted(hash []byte, val interface{}) {
```

Comment:
are you sure this is used only in tests ?

Reply:
yes